src/asm3/event.py

```diff
@@ -11,3 +19,39 @@ def get_event(dbo, eventid):
     e = dbo.first_row(dbo.query(get_event_query(dbo) + " WHERE ev.ID = %d" % eventid))
     return e
 
+
+def insert_event_from_form(dbo, post, username):
+    l = dbo.locale
+    pid = dbo.get_id("event")
+    ownerid = post["ownerid"]
+    if ownerid == "" or ownerid == "0":
+        ownerid = None
+
+    if post["startdate"].strip() == "":
+        raise asm3.utils.ASMValidationError(_("Event must have a start date.", l))
+
+    if post["enddate"].strip() == "":
+        raise asm3.utils.ASMValidationError(_("Event must have an end date.", l))
+
+    if post["address"].strip() == "":
+        raise asm3.utils.ASMValidationError(_("Event must have an address.", l))
+
+    if post.date("startdate") > post.date("enddate"):
+        raise asm3.utils.ASMValidationError(_("End date must be equal to or later than start date."))
+
+    dbo.insert("event", {
+        "ID": pid,
+        "StartDateTime": post.date("startdate"),
+        "EndDateTime": post.date("enddate"),
+        "EventName": post["eventname"],
+        "EventOwnerID": ownerid,
+        "EventAddress": post["address"],
+        "EventTown": post["town"],
+        "EventCounty": post["county"],
+        "EventPostCode": post["postcode"],
+        "EventCountry": post["country"]
+    }, user=username, generateID=False)
```

Comment:
The new ID is the return value of dbo.insert - there's no need to specify the ID in the values list or call getid above, or set generateID=False.  Just do pid = dbo.insert("event", ...